Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,236 @@ public static class BalanceStats
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+                
+                if (isBalanceValid && balance >= maxBalance)
+                {
+                    if (balance > maxBalance)
+                    {
+                        name = "";
+                    }
+                    if (name != "")
+                    {
+                        name += ", ";
+                    }
+                    name += peopleBalances[0];
+
+                    maxBalance = balance;
+                }
+            }
+
+            SetMoneyFormat();
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word = name.Contains(", ") ? "are" : "is";
+            var word2 = name.Contains(", ") ? "people" : "person";
+            return $"{ReplaceLastComma(name)} {word} the richest {word2}. {maxBalanceNoComma}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var minBalance = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid && balance <= minBalance)
+                {
+                    if (balance < minBalance)
+                    {
+                        name = "";
+                    }
+                    if (name != "")
+                    {
+                        name += ", ";
+                    }
+                    name += peopleBalances[0];
+
+                    minBalance = balance;
+                }
+            }
+
+            var word = name.Contains(", ") ? "have" : "has";
+            SetMoneyFormat();
+            var minBalanceNoComma = RemoveComma($"{minBalance:C0}").Replace("(", "-").Replace(")", "");
+            if (minBalance < 0)
+            {
+                minBalanceNoComma = $"{minBalanceNoComma}";
+            }
+            return $"{ReplaceLastComma(name)} {word} the least money. {minBalanceNoComma}.";
+        }
+
+        private static string ReplaceLastComma(string name)
+        {
+            var lastComma = name.LastIndexOf(", ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)